demisto_sdk/commands/common/hook_validations/incident_type.py

```diff
@@ -119,3 +120,39 @@ def is_valid_playbook_id(self):
             if self.handle_error(error_message, error_code, file_path=self.file_path):
                 return False
         return True
+
+    def is_valid_autoextract(self):
+        auto_extract_data = self.current_file.get('extractSettings', {}).get('fieldCliNameToExtractSettings')
+
+        # no auto extraction set in incident type.
+        if not auto_extract_data:
+            return True
+
+        invalid_incident_fields = []
+        for incident_field in auto_extract_data:
+            extracting_all = auto_extract_data[incident_field].get('isExtractingAllIndicatorTypes')
+            extract_as_is = auto_extract_data[incident_field].get('extractAsIsIndicatorTypeId')
+            extracted_indicator_types = auto_extract_data[incident_field].get('extractIndicatorTypesIDs')
+
+            # General format check.
+            if type(extracting_all) != bool or type(extract_as_is) != str or type(extracted_indicator_types) != list:
+                invalid_incident_fields.append(incident_field)
+
+            # If trying to extract without regex make sure extract all is set to
+            # False and the extracted indicators list is empty
+            elif extract_as_is != "":
```

Comment:
```suggestion
            elif extract_as_is:
```

Reply:
I kinda wanna be explicit here in order for this to be more understanding about what is being checked - even if it looks kinda redundant.